src/fu-engine.c

```diff
@@ -4136,6 +4136,7 @@ fu_engine_update_history_device (FuEngine *self, FuDevice *dev_history, GError *
 			const gchar *csum = g_ptr_array_index (checksums, i);
 			fu_device_add_checksum (dev_history, csum);
 		}
+		fu_device_set_version (dev_history, fu_device_get_version (dev));
```

Comment:
What is the pending `version_new` and `version_old` values after the update has been applied and the user has rebooted? i.e. before `fu_engine_update_history_device()` runs? I think this might be papering over a bigger bug that needs some investigation first.

Reply:
I cleared pending.db and I re-ran with the stuff in #1143 (after my fu-device donor fix for version format) but without my extra change in this PR.

Pending.db doesn't get updated at all.  Exact same content before and after reboot.
```
$ sqlite3 pending.db
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
sqlite> .headers on
sqlite> select * from history;
device_id|update_state|update_error|filename|display_name|plugin|device_created|device_modified|checksum|flags|metadata|guid_default|version_old|version_new|checksum_device|protocol
5dbeb140337f610d54913c0f43d15fdaa0eafb51|4|||Micron 1300 SATA 256GB|ata|1554755186|0|722a4651af36cf770135e03d99193b826af8d0ac|1573131|CpuArchitecture=x86_64;RuntimeVersion(org.freedesktop
.fwupd)=1.2.7;DistroId=ubuntu;CompileVersion(org.freedesktop.fwupd)=1.2.7;CompileVersion(com.redhat.fwupdate)=12;CompileVersion(org.freedesktop.gusb)=0.2.11;RuntimeVersion(org.freedesktop
.appstream-glib)=0.7.14;KernelVersion=4.15.0-29-generic;DistroVersion=18.04;CompileVersion(com.redhat.efivar)=34;RuntimeVersion(com.dell.libsmbios)=2.4;BootTime=1554488415;RuntimeVersion(
com.redhat.fwupdate)=12|cdfcae10-d678-5e2b-9742-d35891619fd2|13000000|13000001||
```

Reply:
Seeing `|13000000|13000001|`... shouldn't we be saving the post-processed version in the history database? e.g. `0.198.23872`?

Reply:
Quick question; are you assuming that `fu_device_set_version()` is converting the format from `13000000` to `0.198.23872`? So far I've not done that, and the plugins have done that if they needed to...

Reply:
I wouldn't expect that `fu_device_set_version` did conversion on it's own, but commits like https://github.com/hughsie/fwupd/commit/4bae7a2fa864fd6ff8091539fbaf896e1edf35ae were done to be impervious to the format being post-processed or not when compared.

Reply:
I was under the assumption the version shouldn't be converted at all when the version_format was plain.

Reply:
the message `installed version %s matching history %s` does match the "unprocessed" version though too.

Reply:
Wait, so you the version is expected to be shown as `13000000` and `0.198.23872` is the **wrong** format?

Reply:
Yeah, I mean it's a plain version format, so that's what I expect to see and be used everywhere.  It's the format I see with get-devices, what I put in the XML file and what I expected to see in the history db (and am seeing right now)

Reply:
Okay, so that was a little wild-goose chase on my part then :)